pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -97,13 +97,21 @@ func (sm SetMatcher) Matches(obj runtime.Object) (bool, error) {
 	return sm.Has(pod.Name), nil
 }
 
-// EverythingMatcher matches everything
-type EverythingMatcher struct{}
+func (setMatcher) MatchesSingle() (string, bool) {
+	return "", false
```

Comment:
Why would that be guaranteed to match a single resource?  Field "portalIP=None" would have a single selector but match many services.


> On Apr 3, 2015, at 2:11 AM, Wojciech Tyczynski <notifications@github.com> wrote:
> 
> In pkg/registry/generic/etcd/etcd_test.go:
> 
> > @@ -97,13 +97,21 @@ func (sm SetMatcher) Matches(obj runtime.Object) (bool, error) {
> >  	return sm.Has(pod.Name), nil
> >  }
> >  
> > -// EverythingMatcher matches everything
> > -type EverythingMatcher struct{}
> > +func (setMatcher) MatchesSingle() (string, bool) {
> > +	return "", false
> I think it should return true if the size of StringSet is set to 1.
> 
> —
> Reply to this email directly or view it on GitHub.
>